In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency

In [2]:
df=pd.read_csv("/content/BuyerRatio (1).csv")
df

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [3]:
#make dimensional array
obs=np.array([[50,142,131,70],[435,1523,1356,750]])
obs


array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [4]:
#chi2_contingency independace test
chi2_contingency(obs)  # o/p is (chi2 statsvalue , p_value, df, expected obsevation )

Chi2ContingencyResult(statistic=1.595945538661058, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [5]:
# compare p_value with α = 0.05


Inferrence: As (p_value = 0.6603) > (α = 0.05); Accept the null hypothesis i.e independence of categorical variable thus ,male female buyer reations are similar across regions and are not related